src/services/book-service.ts

```diff
@@ -0,0 +1,115 @@
+import { Request } from 'express';
+import { Document, Query, Schema, SortOrder } from 'mongoose';
+
+import { Book, IBook } from '../models/book';
+import { IBookQuery } from '../interfaces/book-query';
+import { UserRoleEnum } from '../utils/common/enum';
+import {
+  COMMON_QUERIES,
+  BOOK_NON_STRING_QUERIES
+} from '../utils/common/book-filters';
+
+import { toLowerCaseQuery } from '../utils/functions/lower-case-query';
+
+class BookService {
+  public async editSpecificBook(
+    bookId: string,
+    userId: Schema.Types.ObjectId | undefined,
+    updatedBook: IBook
+  ): Promise<Document<any, any, IBook> | null> {
+    return Book.findOneAndUpdate(
+      {
+        _id: bookId,
+        user_id: userId
+      },
+      updatedBook,
+      { new: true }
+    );
+  }
+
+  public async deleteSpecificBook(
```

Comment:
Cẩn thận delete có khi là từ khoá, nếu không được thì dùng remove

```suggestion
  public async delete(
```